third-culture-software/bhima
# Add budget module

This PR adds a new module under the Finance menu for defining and tracking a budget.

Closes https://github.com/IMA-WorldHealth/bhima/issues/7086
Closes https://github.com/IMA-WorldHealth/bhima/issues/2807
Closes https://github.com/IMA-WorldHealth/bhima/issues/267  

Note that I had not seen Issue https://github.com/IMA-WorldHealth/bhima/issues/267 before today, but I think the new budget module implements most of what the issue describes.

*Preliliminary Documention*
- [BHIMA Budget Module (English) v5.pdf](https://github.com/IMA-WorldHealth/bhima/files/13442234/BHIMA.Budget.Module.English.v5.pdf)
- [Module Budgétaire BHIMA (French) v5.pdf](https://github.com/IMA-WorldHealth/bhima/files/13442247/Module.Budgetaire.BHIMA.French.v5.pdf)  Thanks for @Ebed-meleck for editing the French version!




**TESTING**

Here is how to test the basics of the budget module:
- Use bhima_test 
- Run `npm run build:db' to update the database_
- There is very little "Actuals" data in the test database, so do the next step to provide a little data
- Go to the `Finance > Journal`.  Select all transactions.   Click on the [Trial Balance] button on the toolbar and click on the [Submit] button in the Trial Balance dialog.
- Go to the `Administration > Role Management` page and edit your user permissions to allow the user to access the `Finance > Budget Management` page.
- Go to the `Finance > Budget Management` page.  You should see a display of the accounts with the budget and actuals data for the fiscal year.  
- Note that you can select a different fiscal year at the bottom of the page.
- In the [Menu] drop-down, select the 'Export YTD Budget Data (CSV)' option.  After the file has been created, edit the Budget column on any of the accounts you wish.  Note that there are only a couple of accounts that have actuals data.
- Go back to the BHIMA budget page.  In the [Menu], click on the 'Import Budget CSV file" option.
- You should see the display with added budget data.
- You can use the [Menu] to export a report with the current status
- You should also run the tests (`npm run test`) to verify that the basic tests work (which includes several tests related to the budget module.
